### PR TITLE
EWPP-468: Add sections to site_specific_footer.

### PR DESCRIPTION
**Pull request by drishu**
### Description

Add sections to site_specific_footer.

### Change log

- Added:
- Changed:
   - adapt FooterBlockBase to sections.
   - split getSiteSpecificFooterLinks into 2 new methods.
   - adapt templates to sections.

### CODE REVIEW

File: src/FooterLinkManager.php [hunk 1]
```diff
@@ -81,4 +81,14 @@ public function getSectionsAsOptions(): array {
...
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getListCacheTags(): array {
+    return array_merge(
+      $this->linkStorage->getEntityType()->getListCacheTags(),
+      $this->sectionStorage->getEntityType()->getListCacheTags()
+    );
+  }
```
Comment (ademarco) [inline src/FooterLinkManager.php]: Let's remove this form here, it's not this service's scope to return cache tags of two entity type storage.

File: src/FooterLinkManagerInterface.php [hunk 2]
```diff
@@ -44,4 +44,15 @@ public function getLinksWithoutSection(): array;
...
+  /**
+   * The list cache tags associated with this entity type.
+   *
+   * Enables code listing entities of this type to ensure that newly created
+   * entities show up immediately.
+   *
+   * @return array
+   *   The list cache tags.
+   */
+  public function getListCacheTags(): array;
```
Comment (ademarco) [inline src/FooterLinkManagerInterface.php]: Let's remove this form here, it's not this service's scope to return cache tags of two entity type storage.

File: src/FooterLinkManager.php (hunk 1 above)
Comment (ademarco) [reply src/FooterLinkManager.php]: Also mind the way tags should be merged, it's not done with an array merge, Drupal offers a service method for that.

File: src/Plugin/Block/FooterBlockBase.php [hunk 3]
```diff
@@ -90,18 +101,49 @@ protected function setSiteSpecificFooter(array &$build, CacheableMetadata &$cach
...
-   *   Type of configs.
    * @param \Drupal\Core\Cache\CacheableMetadata $cache
    *   The Cacheable metadata abject.
    *
    * @return array
    *   The array of links.
    */
-  protected function getSiteSpecificFooterLinks(string $type, CacheableMetadata &$cache): array {
+  protected function getGeneralFooterLinks(CacheableMetadata &$cache): array {
+    $cache->addCacheTags($this->linkManager->getListCacheTags());
```
Comment (ademarco) [inline src/Plugin/Block/FooterBlockBase.php]: Simple add entity type cache tags from the different storages here by calling `addCacheTags` twice, as done below in getSocialFooterLinks().

**Review by ademarco — CHANGES_REQUESTED** [review]

File: src/Plugin/Block/FooterBlockBase.php [hunk 4]
```diff
@@ -90,18 +101,53 @@ protected function setSiteSpecificFooter(array &$build, CacheableMetadata &$cach
...
    */
-  protected function getSiteSpecificFooterLinks(string $type, CacheableMetadata &$cache): array {
+  protected function getGeneralFooterLinks(CacheableMetadata &$cache): array {
+    /** @var \Drupal\Core\Entity\EntityStorageInterface $storage */
+    $storage = $this->entityTypeManager->getStorage('footer_link_general');
+    $cache->addCacheTags($storage->getEntityType()->getListCacheTags());
+    $storage = $this->entityTypeManager->getStorage('footer_link_section');
+    $cache->addCacheTags($storage->getEntityType()->getListCacheTags());
+    $links = [];
+    foreach ($this->linkManager->getSections() as $section) {
```
Comment (ademarco) [inline src/Plugin/Block/FooterBlockBase.php:117]: Let's only return sections with links.

**Review by ademarco — CHANGES_REQUESTED** [review]

**Review by 22Alexandra — APPROVED** [review]

**Review by ademarco — APPROVED** [review]

**Pull request merged**